cli/src/clojure_lsp/server.clj

```diff
@@ -459,7 +494,7 @@
   ;; about which messages are sent when probably needs to be handled in lsp4clj.
   ;; https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#initialize
   (handler/initialize components
-                      (:root-uri params)
+                      (normalize-uri (:root-uri params))
```

Comment:
I'd like some help thinking about this change. I think that this means that the root-uri never contains `%3a` anymore, which means that `:encode-colons-in-path?` is always `false`, and therefore should be removed.

@LuisThiamNye you introduced `:encode-colons-in-path?` in #439. I don't know if you still follow this project, but if you do, can I recruit your help? Do you remember when you saw colons encoded as `%3a` in the root-uri? What OS/editor/client/repo combination was it? Can you test this branch there?

I think `:encode-colons-in-path?` was for Windows, but I don't have access to a Windows machine. @ikappaki, you've done the most recent Windows maintenance. Do you have any insight or concerns about this change?

Apologies for bringing you both in without much context. As a really quick summary, the intention of this line and the other changes to server.clj is to put every URI we receive from a client into a format that A) will match the URIs we get from clj-kondo, and B) can be used by the editors to open files, especially jar files. By doing this uniformly in server.clj, we don't need as much special logic in other places to convert URIs into particular formats.

@ericdallo I did all my testing with `(normalize-uri (:root-uri params))`, so I know it works on a Mac with emacs-lsp, Calva and coc.nvim. If we decide to revert this line, we have to remember to test all those again.

Reply:
👍

I guess project-uri will never come with colons but I may be wrong.

Reply:
> Do you remember when you saw colons encoded as `%3a` in the root-uri?

According to [this comment](https://github.com/clojure-lsp/clojure-lsp/pull/439#issuecomment-851614968), I observed this with Calva on Windows to denote the root drive (`c:` → `c%3A`).

Reply:
Thanks @LuisThiamNye. The unit tests [already cover](https://github.com/clojure-lsp/clojure-lsp/pull/1327/files#diff-8c9f99b87bb2d221ae57cf20e622aac544c6b1b813f227be77c6c7784783b070R234) `c:` being encoded as `c%3a`. I hope that means Calva will work on Windows, but if you or @ikappaki have a chance to confirm, I'd appreciate it.

Reply:
Hi @mainej,

I had a quick play with this branch and Calva, and navigation from the clojure-lsp project to external libs appears to be working as normal. Do you have a particular test case you want us to try out?

Amongst other things, I've tried the following which I think is relevant since I'm jumping into jars:
1. Open `cli/src/lcojure_lsp/server.clj` and right click on `timbre`, goto definition, which gets me into the timbre-5.2.1.jar
2. Right click on `reduce` keyword, goto definition, it gets me into the clojure `core.clj` jar.

Thanks

Reply:
Hi @mainej,

I confirm I can jump from `cli/src/lcojure_lsp/server.clj` -> `~\.m2\repository\com\taoensso\timbre\5.2.1\timbre-5.2.1.jar!\taoensso\timbre.cljc` -> `~\.m2\repository\com\taoensso\timbre\5.2.1\timbre-5.2.1.jar!\taoensso\timbre\appenders\core.cljc` with no issue.

Reply:
Brilliant! Thanks @ikappaki :pray: